### PR TITLE
link fec is now an optional setting

### PR DESCRIPTION
**Pull request by Nieuwejaar**
We no longer require that FEC be specified when creating a new link, instead letting dendrite choose a default value based on the transceiver type.  This PR adapts to that change in the dendrite API and allows a NULL FEC value to be persisted to the database.

### CODE REVIEW

File: schema/crdb/dbinit.sql [hunk 1]
```diff
@@ -2840,11 +2840,11 @@ CREATE TYPE IF NOT EXISTS omicron.public.switch_link_speed AS ENUM (
 );
 
 CREATE TABLE IF NOT EXISTS omicron.public.switch_port_settings_link_config (
-    port_settings_id UUID,
-    link_name TEXT,
-    mtu INT4,
+    port_settings_id UUID NOT NULL,
```
Comment (Nieuwejaar) [inline schema/crdb/dbinit.sql:2843]: These changes aren't strictly related to the FEC part of this PR.  These fields should always have been `NOT NULL`, and that oversight came to light when I realized that the `fec` field was already nullable.

File: schema/crdb/make-link-settings-non-nullable/up1.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE omicron.public.switch_port_settings_link_config ALTER COLUMN speed SET NOT NULL;
+ALTER TABLE omicron.public.switch_port_settings_link_config ALTER COLUMN mtu SET NOT NULL;
+ALTER TABLE omicron.public.switch_port_settings_link_config ALTER COLUMN port_settings_id SET NOT NULL;
```
Comment (smklein) [inline schema/crdb/make-link-settings-non-nullable/up1.sql:3]: This looks a *lot* like https://github.com/oxidecomputer/omicron/tree/main/schema/crdb#211-adding-a-new-column-without-a-default-value -- if there are any databases which exist, where any of these values **are** NULL, then we'd fail to apply the schema upgrade and the rack would be stuck.

1. Are we confident that these are non-NULL across **all** deployments? Why?
2. Could we use a default value here, in the case that there **are** NULL values?

File: schema/crdb/dbinit.sql (hunk 1 above)
Comment (smklein) [reply schema/crdb/dbinit.sql:2843]: Is it possible to split this into two PRs? Marking the "rust side" of things as Nullable is an easy change to approve, but altering these columns to be non-NULL is a more involved change

**Review by smklein — APPROVED** [review]

File: common/src/api/external/mod.rs [hunk 3]
```diff
@@ -2514,7 +2514,7 @@ pub struct SwitchPortLinkConfig {
     pub mtu: u16,
 
     /// The forward error correction mode of the link.
-    pub fec: LinkFec,
```
Comment (ahl) [inline common/src/api/external/mod.rs:2517]: The internal API got a nice doc update; can we do the same here?

https://github.com/oxidecomputer/dendrite/pull/1038/files#diff-cb166332b35711d4032d1bbea0e1243b00ed0d61e13bd9b9cd63fe39b6bd9df7

**Pull request merged**